pkg/tfgen/docs.go

```diff
@@ -1201,52 +1203,54 @@ func (p *tfMarkdownParser) parseImports(subsection []string) {
 		}
 	}
 
-	if i, ok := tryParseV2Imports(token, subsection); ok {
+	if i, ok := tryParseV2Imports(token, body); ok {
 		p.ret.Import = i
 		return
 	}
 
-	var importDocString string
-	for _, section := range subsection {
-		if strings.Contains(section, "**NOTE:") || strings.Contains(section, "**Please Note:") ||
-			strings.Contains(section, "**Note:**") {
+	importDocs := new(bytes.Buffer)
+	for _, line := range strings.Split(body, "\n") {
+		if strings.Contains(line, "**NOTE:") || strings.Contains(line, "**Please Note:") ||
+			strings.Contains(line, "**Note:**") {
 			// This is a Terraform import specific comment that we don't need to parse or include in our docs
 			continue
 		}
 
 		// Skip another redundant comment
-		if strings.Contains(section, "Import is supported using the following syntax") {
+		if strings.Contains(line, "Import is supported using the following syntax") {
 			continue
 		}
 
 		// Remove the shell comment characters to avoid writing this line as a Markdown H1:
-		section = strings.TrimPrefix(section, "# ")
+		if l1, ok := strings.CutPrefix(line, "#"); ok {
+			if l2, ok := strings.CutPrefix(l1, " "); len(l1) == 0 || ok {
+				line = l2
+			}
+		}
 
 		// There are multiple variations of codeblocks for import syntax
-		section = strings.Replace(section, "```shell", "", -1)
-		section = strings.Replace(section, "```sh", "", -1)
-		section = strings.Replace(section, "```", "", -1)
+		line = strings.ReplaceAll(line, "```shell", "")
+		line = strings.ReplaceAll(line, "```sh", "")
+		line = strings.ReplaceAll(line, "```", "")
 
-		if strings.Contains(section, "terraform import") {
+		if strings.Contains(line, "terraform import") {
 			// First, remove the `$`
-			section := strings.Replace(section, "$ ", "", -1)
+			section := strings.ReplaceAll(line, "$ ", "")
 			// Next, remove `terraform import` from the codeblock
-			section = strings.Replace(section, "terraform import ", "", -1)
-			importString := ""
-			parts := strings.Split(section, " ")
-			for i, p := range parts {
+			section = strings.ReplaceAll(section, "terraform import ", "")
+			var name, id string
+			for i, p := range strings.Split(section, " ") {
 				switch i {
 				case 0:
 					if !isBlank(p) {
 						// split the string on . and take the last item
 						// this gets the identifier broken from the tf resource
 						ids := strings.Split(p, ".")
-						name := ids[len(ids)-1]
-						importString = fmt.Sprintf("%s %s", importString, name)
```

Comment:
Why's importString gone now?

Reply:
`importString` has been replaced with `importDocs`, since that's what `emitImportCodeBlock` already operated on.